test/modules/GenericMqttTester/src/message_handler.rs

```diff
@@ -0,0 +1,66 @@
+use mpsc::UnboundedSender;
+use tokio::{sync::mpsc, task::JoinHandle};
+
+use mqtt3::{PublishHandle, ReceivedPublication};
+
+use crate::{MessageTesterError, ShutdownHandle};
+
+/// Responsible for receiving publications and taking some action.
+pub trait MessageHandler {
+    /// Starts handling messages sent to the handler
+    fn run(self) -> (JoinHandle<Result<(), MessageTesterError>>, ShutdownHandle);
+
+    /// Sends a publication to be handled by the message handler
+    fn publication_sender_handle(&self) -> UnboundedSender<ReceivedPublication>;
+}
+
+/// Responsible for receiving publications and reporting result to the TRC.
+pub struct ReportResultMessageHandler {}
+
+impl ReportResultMessageHandler {
+    pub fn new() -> Self {
+        todo!()
+    }
+}
```

Comment:
I am not yet sure how the report result handler will operate so skipping new() impl